### PR TITLE
list of teachers

### PR DESCRIPTION
**Pull request by Tarkeasy**

### CODE REVIEW

File: src/app/admin-panel/admin-panel/teachers/teachers.component.html [hunk 1]
```diff
@@ -1,3 +1,86 @@
...
-</p>
+<p-table [columns]="columns" [value]="teachers" selectionMode="single" [(selection)]="selectedTeacher" (onRowSelect)="onRowSelect($event)"
+    [paginator]="true" [rows]="15">
+    <ng-template pTemplate="caption">
+        Список учителів
+    </ng-template>
+    <ng-template pTemplate="header" let-columns>
+        <tr>
+            <th *ngFor="let col of columns" [pSortableColumn]="col.field">
+                {{col.header}}
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/teachers/teachers.component.html]: indentation

File: src/app/admin-panel/admin-panel/teachers/teachers.component.ts [hunk 2]
```diff
@@ -1,15 +1,43 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit } from "@angular/core";
+import { TeachersService } from "../../../services/teachers.service";
+import { Iteachers } from "src/app/services/teachers";
 
 @Component({
-  selector: 'app-teachers',
-  templateUrl: './teachers.component.html',
-  styleUrls: ['./teachers.component.scss']
+  selector: "app-teachers",
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/teachers/teachers.component.ts]: what the reason of this change?

File: src/app/services/teachers.service.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Iteachers } from './teachers';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class TeachersService {
+   url:string = "https://fierce-shore-32592.herokuapp.com/teachers";
```
Comment (dmitryy90) [inline src/app/services/teachers.service.ts]: format file

File: src/app/services/teachers.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export interface Iteachers {
+  avatar: "string";
+  dateOfBirth: "string";
+  email: "string";
+  firstname: "string";
+  id: 0;
```
Comment (dmitryy90) [inline src/app/services/teachers.ts]: why do you predefine id?
it is must be not required

Comment (Tarkeasy) [reply src/app/services/teachers.ts]: this format from example value 
https://fierce-shore-32592.herokuapp.com/swagger-ui.html#!/teacher-controller/getallUsingGET

File: src/app/admin-panel/admin-panel/teachers/teachers.component.ts (hunk 2 above)
Comment (Tarkeasy) [reply src/app/admin-panel/admin-panel/teachers/teachers.component.ts]: seems weird, don't remember when i did it, there is no reason

**Pull request closed without merging**